pkg/metricshandler/metrics_handler.go

```diff
@@ -178,15 +178,16 @@ func (m *MetricsHandler) Run(ctx context.Context) error {
 	return ctx.Err()
 }
 
-// ServeHTTP implements the http.Handler interface. It writes all generated
-// metrics to the response body.
+// ServeHTTP implements the http.Handler interface. It writes all generated metrics to the response body.
+// Note that all operations defined within this procedure are performed at every request.
 func (m *MetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	m.mtx.RLock()
 	defer m.mtx.RUnlock()
 	resHeader := w.Header()
 	var writer io.Writer = w
 
 	contentType := expfmt.NegotiateIncludingOpenMetrics(r.Header)
+	gotContentType := contentType
```

Comment:
Why do we use a copy of the content type here? If the content type of the request was OpenMetrics and we fallback to Text, I would expect us to propagate Text and not OpenMetrics since that is the format we are going to serve.

Reply:
It seems out of all the `content-type`s (as mentioned below):
- `FmtUnknown`
- `FmtText`
- `FmtProtoDelim`
- `FmtProtoText`
- `FmtProtoCompact`
- `FmtOpenMetrics_1_0_0`
- `FmtOpenMetrics_0_0_1`

We fallback to `FmtText` if the case is anything other than the last two:
```go
if contentType != expfmt.FmtOpenMetrics_1_0_0 && contentType != expfmt.FmtOpenMetrics_0_0_1 {
	contentType = expfmt.FmtText
}
```

But IIUC, since we now have a workaround to handle other `protobuf`-based types as well, we can drop this fallback behavior.

Reply:
Sent a patch.

Reply:
My original comment was confusing sorry. I meant if the content type of the request is **anything other** (I wrote the opposite above sry) than OpenMetrics, we would want to propagate the new FmtProtoText down to the parser, when what you did before was to propagate the content type of the request that you stored in `gotContentType`.

So my suggestion was just to propagate `contentType` directly to the parser.